docs/topics/db/transactions.txt

```diff
@@ -252,6 +252,130 @@ by Django or by third-party libraries. Thus, this is best used in situations
+
+.. function:: on_commit(func, using=None)
+
+Pass any function (that takes no arguments) to :func:`on_commit`::
+
+    from django.db import transaction
+
+    def do_something():
+        pass  # send a mail, invalidate a cache, fire off a Celery task, what-have-you.
+
+    transaction.on_commit(do_something)
+
+You can also wrap your thing up in a lambda::
+
+    transaction.on_commit(lambda: some_celery_task.delay('arg1'))
+
+The function you pass in will be called immediately after a hypothetical
+database write made at the same point in your code is successfully
+committed.
+
+If you register a callback while there is no transaction active, it will be
+executed immediately.
+
+If that hypothetical database write is instead rolled back (typically when an
+unhandled exception is raised in an :func:`atomic` block), your function will
+be discarded and never called.
+
+
+Order of execution
+------------------
+
+On-commit functions for a given transaction are executed in the order they were
+registered.
+
+
+Exception handling
+------------------
+
+If one on-commit hook within a given transaction raises an uncaught exception,
+no later-registered hooks in that same transaction will run. (This is, of
+course, the same behavior as if you'd executed the hooks sequentially yourself
+without :func:`on_commit`.)
+
+
+Timing of execution
+-------------------
+
+Your hook functions are executed *after* a successful commit, so if they fail,
+it will not cause the transaction to roll back. They are executed conditionally
+upon the success of the transaction, but they are not *part* of the
+transaction. For the intended use cases (mail notifications, Celery tasks,
+etc), this is probably fine. If it's not (if your follow-up action is so
+critical that its failure should mean the failure of the transaction itself),
+then you don't want to use the :func:`on_commit` hook. (Instead, you may want
+`two-phase commit`_.)
+
+.. _two-phase commit: http://en.wikipedia.org/wiki/Two-phase_commit_protocol
+
+Use in tests
+------------
+
+Django's :class:`~django.test.TestCase` class wraps each test in a transaction
+and rolls back that transaction after each test, in order to provide test
+isolation. This means that no transaction is ever actually committed, thus your
+:func:`on_commit` hooks will never be run. If you need to test the results of
+an :func:`on_commit` hook, you may need to use
+:class:`~django.test.TransactionTestCase` instead.
+
+
+Savepoints
+----------
+
+Savepoints (i.e. nested :func:`atomic` blocks) are handled correctly. That is,
+an :func:`on_commit` hook registered after a savepoint (in a nested
+:func:`atomic` block) will be called after the outer transaction is committed,
+but not if a rollback to that savepoint or any previous savepoint occurred
+during the transaction.
+
+Use autocommit and transaction.atomic
+-------------------------------------
+
+.. warning::
+    On-commit hooks only works correctly in :ref:`autocommit
```

Comment:
I'm not sure that this paragraph really needs to be called out as a warning block, especially since trying to do what it cautions against will result in a clear error anyway, not silent confusing behavior.

There are also some subject-verb agreement issues in this sentence. Suggested rewrite:

> On-commit hooks only work with autocommit mode and the atomic (or ATOMIC_REQUESTS) transaction API. Calling on_commit when autocommit is disabled and you are not within an atomic block will result in an error.